### PR TITLE
SpinButton: remove default min and max

### LINKED ISSUE #11358: SpinButton default max of 100, misleading optional property
### Environment Information

- **Package version(s)**: `office-ui-fabric-react@7.59.0`
- **Browser and OS versions**: Google Chrome is up to date
Version 78.0.3904.108 (Official Build) (64-bit)

### Please provide a reproduction of the bug in a codepen:

There is misleading documentation.
The docs claim the default is 10, but it appears to be 100.
The types also make the property optional implying that if you don't supply a `max` it will have no max value other than the assumed MAX_INTEGER or something.

The issue is that not giving a `max` property means the `max` is 100. 

See codepen. Taken from https://developer.microsoft.com/en-us/fabric#/controls/web/spinbutton then removed all extra unecessary code related to bug.

Just type in 2343245 into the input and press tab. It sets to 100.

https://codepen.io/mattmazzola-the-reactor/pen/KKwwgKN?editors=1010

#### Actual behavior:

Spin button limits input to 100 when it shouldn't.  There is other weird behavior where if you type a large number and click outside of the input onto the webpage background the number stays unmodified, but if you enter 2342345 and press tab, enter, or click on another Office Fabric control the input is reset to 100.

This change in behavior based on type of Dismissal reminds me of another bug related to picker dimissal changing. https://github.com/OfficeDev/office-ui-fabric-react/issues/10892

My work around seems to be forcing highest max possible.
`max={Number.MAX_SAFE_INTEGER}`

Update: looks like issue is same with `min` property also limiting to 0

#### Expected behavior:

Number should be not have a `max` if the `max` property is not provided.

### Priorities and help requested:

Are you willing to submit a PR to fix? (Yes, **No**)

Requested priority: (Blocking, High, **Normal**, Low)

Products/sites affected: (if applicable)
Conversation Learner

### PR DESCRIPTION
**Pull request by ecraig12345**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #11358
- [x] Include a change request file using `$ yarn change`

#### Description of changes

Remove the defaults of `min=0` and `max=100` from SpinButton, so that it's unbounded by default. This makes it behave more consistently with the built-in `<input type="number"/>` and likely is more consistent with user expectations.

Other minor changes:
- Revert native props extension back to using `div` props (not `input` props). `div` is the correct element type based on where the native props are actually applied, and it's the element type that was used in v7.
- De-duplicate rendering of the label
- Mark myself as codeowner instead of @khmakoto since I've worked on SpinButton so much recently

### CODE REVIEW

File: packages/react-internal/src/components/SpinButton/SpinButton.base.tsx [hunk 1]
```diff
@@ -156,14 +159,20 @@ export const SpinButtonBase: React.FunctionComponent<ISpinButtonProps> = React.f
...
+    // default validation handling
+    let newNumber = Number(newValue);
+    if (newValue && newValue.trim().length && !isNaN(newNumber)) {
+      if (typeof min === 'number') {
+        newNumber = Math.max(min, newNumber);
+      }
+      if (typeof max === 'number') {
+        newNumber = Math.min(max, newNumber);
+      }
+      return String(newNumber);
```
Comment (khmakoto) [inline packages/react-internal/src/components/SpinButton/SpinButton.base.tsx:171]: So, in this case, would it return `undefined` in scenarios that don't enter that if statement?

**Review by khmakoto — APPROVED** [review]

File: packages/react-internal/src/components/SpinButton/SpinButton.base.tsx (hunk 1 above)
Comment (ecraig12345) [reply packages/react-internal/src/components/SpinButton/SpinButton.base.tsx:171]: Yes, I think that's the same behavior as before, and in this context it means the update is invalid and should be ignored.

Comment (ecraig12345) [reply packages/react-internal/src/components/SpinButton/SpinButton.base.tsx:171]: Oh wait actually it's different (bad interaction of this change with another one). I'll fix that.

Comment (ecraig12345) [reply packages/react-internal/src/components/SpinButton/SpinButton.base.tsx:171]: Hmm never mind, while the behavior of that particular function is slightly different, the overall behavior is still the same. So I'm going to leave it as-is.

**Review by dzearing — APPROVED** [review]

**Pull request merged**